src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -56,6 +57,80 @@ pub struct TargetInfo {
     pub rustdocflags: Rc<[String]>,
 }
 
+/// Check Config (aka `--print=check-cfg` representation)
+#[derive(Debug, Default, Clone)]
+pub struct CheckCfg {
+    /// Is --check-cfg activated
+    pub exhaustive: bool,
+    /// List of expected cfgs
+    pub expecteds: HashMap<String, ExpectedValues<String>>,
+}
+
+/// List of expected check-cfg values
+#[derive(Debug, Clone)]
+pub enum ExpectedValues<T> {
+    /// List of expected values
+    Some(HashSet<Option<T>>),
+    /// All values expected
+    Any,
+}
+
+impl CheckCfg {
+    fn process_line(&mut self, line: &str) {
+        if line == "any()=any()" || line == "any()" {
+            self.exhaustive = false;
+        } else {
+            if let Some((name, val)) = line.split_once('=') {
+                let name = name.to_string();
+
+                // strip "" around the value, e.g. "linux" -> linux
+                let val = val
+                    .strip_prefix('"')
+                    .map(|v| v.strip_suffix('"'))
+                    .flatten()
+                    .unwrap_or(val);
+
+                self.expecteds
+                    .entry(name)
+                    .and_modify(|v| match v {
+                        ExpectedValues::Some(_) if val.is_empty() => {}
+                        ExpectedValues::Some(_) if val == "any()" => {
+                            *v = ExpectedValues::Any;
+                        }
+                        ExpectedValues::Some(v) => {
+                            v.insert(Some(val.to_string()));
+                        }
+                        ExpectedValues::Any => {}
+                    })
+                    .or_insert_with(|| {
+                        if val.is_empty() {
+                            // no value to add
+                            ExpectedValues::Some(Default::default())
```

Comment:
I feel like I'd prefer us to be consistent on whether we use `None` or `Default::default()` so we can more easily correlate the code